src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -512,6 +519,42 @@ nf_task_entry() {
   # run the task
   bash .command.run > >(tee .command.log) 2>&1
   export exit_code=$?
+  kill "$AWS_RELOGIN_PID"
   dx set_properties ${DX_JOB_ID} nextflow_exit_code=$exit_code
   set -e
 }
+
+aws_login() {
+  if [ -f "$AWS_ENV" ]; then
+    source $AWS_ENV
+    # aws env file example values:
+    # "iamRoleArnToAssume", "roleSessionName", "jobTokenAudience", "jobTokenSubjectClaims", "region"
+    python3 -m pip install --upgrade 'git+https://github.com/dnanexus/dx-toolkit.git@master#egg=dxpy&subdirectory=src/python' # FIXME: Remove after it is available on staging.
+    job_id_token=$(dx-jobutil-get-identity-token --aud ${jobTokenAudience} --subject_claims ${jobTokenSubjectClaims})
+    output=$(aws sts assume-role-with-web-identity --role-arn $iamRoleArnToAssume --role-session-name $roleSessionName --web-identity-token $job_id_token --duration-seconds 3600)
+    mkdir -p /home/dnanexus/.aws/
+
+    cat <<EOF > /home/dnanexus/.aws/credentials
+[default]
+aws_access_key_id = $(echo "$output" | jq -r '.Credentials.AccessKeyId')
+aws_secret_access_key = $(echo "$output" | jq -r '.Credentials.SecretAccessKey')
+aws_session_token = $(echo "$output" | jq -r '.Credentials.SessionToken')
+EOF
+    cat <<EOF > /home/dnanexus/.aws/config
+[default]
+region = $region
+EOF
+  else
+    echo "No AWS environment variables available"
+  fi
+}
+
+automatic_aws_relogin() {
+  while true; do
+      sleep 3000 # 50 minutes, first login is done independently, so we wait first
+      if [ -f "$AWS_ENV" ]; then
```

Comment:
Would expect file not to exist if AWS login not used for this job, right?

Reply:
Correct, basically, this file is present only if it is created in the headjob.

I thought of not creating the aws_relogin process in case it is not created, but I believe it would create too much code and also we would need to check whether it exists or not. (as for example in the headjob we do not know whether it should exist until the headjob creates the aws_env file.).

And as this thread is mostly sleeping, I dont see the problem of having the additional thread.